src/types/message.ts

```diff
@@ -196,13 +196,20 @@ export interface Application {
   bot?: UserPayload;
 }
 
+/** https://discord.com/developers/docs/resources/channel#message-object-message-reference-structure */
 export interface Reference {
-  /** The id of the originating message */
-  "message_id"?: string;
-  /** The id of the originating message's channel */
-  "channel_id": string;
-  /** The id of the originating message's guild */
-  "guild_id"?: string;
+  /** ID of the originating message */
+  message_id?: string;
+  /**
+   * ID of the originating message's channel
+   * 
+   * NOTE: `channel_id` is optional when creating a reply, but will always be present when receiving an event/response that includes this data model
```

Comment:
This can be handled better imo, we just need to split this into 2 interfaces.

export interface ReferenceBase {

}

export interface Reference extends ReferenceBase {
  channel_id: string;
}

Reply:
Can you explain as to why this is necessary?

Reply:
For a few reasons:
1. accuracy of typings although this isnt that important
2. user experience. with the current solution u have to do `if (data.channel_id)` to make sure a channel id is present before using it but if this typing is more detailed when a user gets it inside an interaction they wont have to leading to cleaner code